### PR TITLE
Add bundled discovery configs to default config.d for rpm and deb packages

### PR DESCRIPTION
**Pull request by rmfitzpatrick**
These changes include updated, commented-out discovery configs to the default `/etc/otel/collector/config.d` directory for rpm and deb packages. They also include updates to the commented header and integration tests for basic discovery mode validation. These tests aren't going to be included in CI out of the gate, but that's my intention in the near future.

cc @aunshc @theletterf

### CODE REVIEW

**Review by atoulme — APPROVED** [review]

File: cmd/otelcol/config/collector/config.d.linux/properties.discovery.yaml.example [hunk 1]
```diff
@@ -0,0 +1,18 @@
+# Copy this file to properties.discovery.yaml and add your uncommented discovery properties for use in discovery mode:
+# For more information:
+# https://github.com/signalfx/splunk-otel-collector/tree/main/internal/confmapprovider/discovery#discovery-properties
```
Comment (theletterf) [inline cmd/otelcol/config/collector/config.d.linux/properties.discovery.yaml.example:3]: Ideally we'll want to link to the official docs after launch, although this one is technical enough to perhaps not require it.

**Review by theletterf — APPROVED** [review]

**Review by jeffreyc-splunk — APPROVED** [review]
The packaging changes LGTM.

File: cmd/otelcol/config/collector/config.d.linux/extensions/docker-observer.discovery.yaml [hunk 2]
```diff
@@ -1,5 +1,10 @@
 #####################################################################################
-# This file is generated by the Splunk Distribution of the OpenTelemetry Collector. #
+#           This file is generated by the Splunk OpenTelemetry Collector.           #
```
Comment (aunshc) [inline cmd/otelcol/config/collector/config.d.linux/extensions/docker-observer.discovery.yaml:2]:
```suggestion
#           This file is generated by the Splunk Distribution of the OpenTelemetry Collector.           #
```
I still recommend using this language for our distro.

File: internal/confmapprovider/discovery/README.md [hunk 3]
```diff
@@ -1,6 +1,6 @@
-# Discovery confmap.Provider (Experimental)
+# Discovery confmap.Provider
 
-**This component should not be considered stable. At this time its functionality is provided for testing and validation purposes only.**
+**This feature is currently in [development](https://github.com/open-telemetry/opentelemetry-collector#development)**
```
Comment (aunshc) [inline internal/confmapprovider/discovery/README.md:3]: Would the alpha status be more appropriate? Let's discuss the existing state and path to the next phases of maturity.

**Pull request merged**